src/EventListener/BulkSavePropertyListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flagbit\Bundle\CategoryBundle\EventListener;
+
+use Akeneo\Pim\Enrichment\Component\Category\Model\CategoryInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Flagbit\Bundle\CategoryBundle\Entity\CategoryProperty;
+use Flagbit\Bundle\CategoryBundle\Repository\CategoryPropertyRepository;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+use function count;
+
+/**
+ * Category post save listener that handles bulk saves with properties.
+ */
+class BulkSavePropertyListener
+{
+    /** @var ParameterBag<string, array<string, mixed>> */
+    private ParameterBag $propertiesBag;
+    private CategoryPropertyRepository $repository;
+    private EntityManagerInterface $entityManager;
+
+    /**
+     * @phpstan-param ParameterBag<string, array<string, mixed>> $propertiesBag
+     */
+    public function __construct(
+        ParameterBag $propertiesBag,
+        CategoryPropertyRepository $repository,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->propertiesBag = $propertiesBag;
+        $this->repository    = $repository;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @phpstan-param GenericEvent<mixed> $event
+     */
+    public function onBulkCategoryPostSave(GenericEvent $event): void
+    {
+        // Save category properties
+        foreach ($event->getSubject() as $category) {
+            if (! $category instanceof CategoryInterface) {
+                continue;
+            }
+
+            if (! $this->propertiesBag->has($category->getCode())) {
+                continue;
+            }
+
+            $properties = $this->propertiesBag->get($category->getCode());
+            if (count($properties) === 0) {
+                return;
+            }
+
+            $categoryProperty = $this->findProperty($category);
+            $categoryProperty->setProperties($properties);
```

Comment:
Akeneo allows to have an import file without all the containing data of an entity. For example: A category that has code and the category name in three languages can import a csv file that only contains the code and one of the names in one language. Not having the other 2 languages for the name don't affect those in the entity itself. Neither are they changed nor deleted.

Setting the properties of such a shortened import file would remove the other properties in the current code when used with `setProperties()`. I'd suggest to add a new method to the entity called `mergeProperties()`.

Reply:
A function has been implemented and can be reviewed / tested.